### ISSUE TITLE
Hang after "Show details for all processes"

### ISSUE THREAD

**Issue by TETYYS** [winsiderss/systeminformer#1755]
### Brief description of your issue

Shortly after turning on my PC I launched SI and pressed "Show details for all processes", when to my surprise no window showed up. After launching a second instance I saw that first instance has hanged with following stack trace

![image](https://github.com/winsiderss/systeminformer/assets/3091645/4be2fa06-0b2b-4d2b-8065-cf8bcdfa0302)

The kernel driver KSystemInformer is also stuck, on status "Start pending", therefore my second instance threw this error at me:

![image](https://github.com/winsiderss/systeminformer/assets/3091645/a845a3cd-eba0-4848-b0e2-771d84bbbefa)

I will not bother restarting to fix the driver service, so if you have any additional steps I can take to debug this, feel free to write some

### Steps to reproduce (optional)

Not sure, usually when I do that new admin-enabled window shows up fine

### Environment (optional)

```shell
System Informer 3.0.6550
Windows 10 22H2 19045.2965
```

**Comment by jxy-s** [comment]
If the system is still in that state. It would be helpful if you captured a kernel mode dump. You're welcome to email me the dump file.

Additionally, if you're able to reproduce the problem capturing a trace of this occurring would be helpful.

1. open an elevated command prompt
2. execute: `logman create trace ksi -p {F64B58A2-8214-4037-8C7D-B96CE6098F3D} 0xffffffff 0xff -bs 64 -nb 16 128 -o "kph.etl" -v mmddhhmm -ow`
3. execute: `logman start ksi`
4. reproduce the issue
5. execute: `logman stop ksi`
6. execute: `logman delete -n ksi`
7. harvest the `kph_nnnnnnnn.etl` trace log from disk
8. attach the log file here or email it to us

It's possible this is an application compatibility problem similar to https://github.com/winsiderss/systeminformer/issues/1515 - we have a fix for this issue on master but haven't been able to release a new nightly due to https://github.com/winsiderss/systeminformer/issues/1750#issuecomment-1582196835.

**Comment by jxy-s** [comment]
If the system is still in that state. It would be helpful if you captured a kernel mode dump. You're welcome to email me the dump file.

Additionally, if you're able to reproduce the problem capturing a trace of this occurring would be helpful.

1. open an elevated command prompt
2. execute: `logman create trace ksi -p {F64B58A2-8214-4037-8C7D-B96CE6098F3D} 0xffffffff 0xff -bs 64 -nb 16 128 -o "kph.etl" -v mmddhhmm -ow`
3. execute: `logman start ksi`
4. reproduce the issue
5. execute: `logman stop ksi`
6. execute: `logman delete -n ksi`
7. harvest the `kph_nnnnnnnn.etl` trace log from disk
8. attach the log file here or email it to us

It's possible this is an application compatibility problem similar to https://github.com/winsiderss/systeminformer/issues/1515 - we have a fix for this issue on master but haven't been able to release a new nightly due to https://github.com/winsiderss/systeminformer/issues/1750#issuecomment-1582196835.

You could also test this option: `Options > Show advanced options > Advanced > KphDisableImageLoadProtection > 1` - restart the driver after enabling that.

**Comment by TETYYS** [comment]
Didn't manage to recreate it, but at least I send the memory dump

**Comment by jxy-s** [comment]
The driver failed driver entry with:

```
4: kd> !error 0n-1073740756
Error code: (NTSTATUS) 0xc000042c (3221226540) - The requested operation requires elevation.
```

And is suck here:
```
4: kd> k
  *** Stack trace for last set context - .thread/.cxr resets it
 # Child-SP          RetAddr               Call Site
00 ffffe28b`7ef250c0 fffff806`1901c9c0     nt!KiSwapContext+0x76
01 ffffe28b`7ef25200 fffff806`1901beef     nt!KiSwapThread+0x500
02 ffffe28b`7ef252b0 fffff806`1901b793     nt!KiCommitThreadWait+0x14f
03 ffffe28b`7ef25350 fffff806`1910626b     nt!KeWaitForSingleObject+0x233
04 ffffe28b`7ef25440 fffff806`190d5f94     nt!ExfWaitForRundownProtectionRelease+0xfb
05 ffffe28b`7ef254b0 fffff806`13d8cc0e     nt!ExWaitForRundownProtectionRelease+0x24
06 ffffe28b`7ef254e0 fffff806`13dd8632     FLTMGR!FltpWaitForRundownProtectionReleaseInternal+0x116
07 ffffe28b`7ef255f0 fffff806`9533124d     FLTMGR!FltUnregisterFilter+0x1a2
08 ffffe28b`7ef256b0 fffff806`95333c3f     SystemInformer!KphFltUnregister+0x15 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\informer_file.c @ 383] 
09 ffffe28b`7ef256e0 fffff806`95333723     SystemInformer!KphpDriverCleanup+0x27 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\main.c @ 80] 
0a ffffe28b`7ef25710 fffff806`95340020     SystemInformer!DriverEntry+0x26b [C:\Users\aione\source\repos\systeminformer\KSystemInformer\main.c @ 362] 
0b ffffe28b`7ef258d0 fffff806`1956c0f4     SystemInformer!GsDriverEntry+0x20 [minkernel\tools\gs_support\kmodefastfail\gs_driverentry.c @ 47] 
0c ffffe28b`7ef25900 fffff806`195382b5     nt!PnpCallDriverEntry+0x4c
0d ffffe28b`7ef25960 fffff806`1957ca97     nt!IopLoadDriver+0x4e5
0e ffffe28b`7ef25b30 fffff806`1908e5c5     nt!IopLoadUnloadDriver+0x57
0f ffffe28b`7ef25b70 fffff806`191265f5     nt!ExpWorkerThread+0x105
10 ffffe28b`7ef25c10 fffff806`192048d8     nt!PspSystemThreadStartup+0x55
11 ffffe28b`7ef25c60 00000000`00000000     nt!KiStartSystemThread+0x28
```

The reason the filter isn't unregistering is because the communication port is still active, that must completely run down before the filter instance can be unregistered. Which should be done before that call:

https://github.com/winsiderss/systeminformer/blob/master/KSystemInformer/informer_file.c#L365-L384
```cpp
/**
 * \brief Unregisters this driver with the mini-filter.
 */
_IRQL_requires_max_(PASSIVE_LEVEL)
VOID KphFltUnregister(
    VOID
    )
{
    PAGED_PASSIVE();

    KphCommsStop();  <<< HERE

    if (!KphFltFilter)
    {
        return;
    }

    FltUnregisterFilter(KphFltFilter);
    KphFltFilter = NULL;
}
```

The call to `KphCommsStop` resets some global state to zero, unconditionally.. however that isn't the case:

```
4: kd> x SystemInformer!KphpMessage*
fffff806`95327200 SystemInformer!KphpMessageLookaside = struct _PAGED_LOOKASIDE_LIST
fffff806`95327380 SystemInformer!KphpMessageQueue = struct _KQUEUE
fffff806`953271e8 SystemInformer!KphpMessageQueueThreads = 0xffffaa8e`1494da20
fffff806`95327300 SystemInformer!KphpMessageQueueItemLookaside = struct _NPAGED_LOOKASIDE_LIST
fffff806`953271f0 SystemInformer!KphpMessageQueueThreadsCount = 0xc
fffff806`9532db50 SystemInformer!KphpMessageQueueThread (void *)
```

Looking at the disassembly for what you have running on your machine:
```
    SystemInformer!KphFltUnregister:
fffff806`95331238 4883ec28         sub     rsp, 28h
fffff806`9533123c 488b0dc561ffff   mov     rcx, qword ptr [SystemInformer!KphFltFilter (fffff80695327408)]
fffff806`95331243 4885c9           test    KphFltFilter (rcx), KphFltFilter (rcx)
fffff806`95331246 740d             je      SystemInformer!KphFltUnregister+0x1d (fffff80695331255)
fffff806`95331248 e84314ffff       call    SystemInformer!FltUnregisterFilter (fffff80695322690)
fffff806`9533124d 488325b361ffff00 and     qword ptr [SystemInformer!KphFltFilter (fffff80695327408)], 0
fffff806`95331255 4883c428         add     rsp, 28h
fffff806`95331259 c3               ret     
fffff806`9533125a cc               int     3
fffff806`9533125b cc               int     3
```

Looks like the call to `KphCommsStop` is not present.

This mishandling of rundown was actually addressed with my commit here: https://github.com/winsiderss/systeminformer/commit/508d2c1dc5189725899124c9dd828769dc80302d

However, since we haven't been able to do another release with my driver fixes, it's hung up there.

With respect to why the driver entry seemingly randomly failed with `STATUS_ELEVATION_REQUIRED`... I have no clue... that's a very strange code to observe in `DriverEntry`. I did a browse through the initialization we do and nothing jumped out at me as to why that would have occurred.

Regardless, I can say that this specific issue should be resolved with the latest driver, as soon as we can do another nightly build you shouldn't encounter that hang again. But you might see the failed driver entry still.

I also want to inform you, in case this isn't intentional on your part. You do have this loaded in your kernel:
```
fffff806`957e0000 fffff806`957e7000   WinRing0x64   (deferred)    
```

**Comment by jxy-s** [comment]
Blocked by: https://github.com/winsiderss/systeminformer/issues/1668

**Comment by TETYYS** [comment]
> I also want to inform you, in case this isn't intentional on your part. You do have this loaded in your kernel:

yeah, that's intentional, thanks

**Comment by jxy-s** [comment]
hang should be resolved by latest nightly, if you run into another failure to load the driver with that error please open a new issue
